File: SmartG-client/src/app/_interfaces/portfolio-for-creation-dto.ts
import { ContentBlockForCreationDto } from "./content-block-for-creation-dto";

export interface PortfolioForCreationDto {
 
    title: string;
    content: string;
    excerpt: string;
    metaDescription: string;
    metaKeyWords: string;
    slug: string;
    sgCategoryId?: any;
    sgImageId?: number;
    sgGalleryId?: number;
    contentBlocks: ContentBlockForCreationDto[];
}

File: SmartG.Shared/DTOs/PortfolioForCreationDto.cs
using System;
using Microsoft.VisualBasic;

namespace SmartG.Shared.DTOs
{
    public class PortfolioForCreationDto:BaseDto
    {
        public PortfolioForCreationDto()
        {
            DateCreated = DateTime.Now;
            DateUpdated = DateAndTime.Now;
            Deleted = false;
        }

        public string? Title { get; set; }
        public string? Content { get; set; }
        public string? Excerpt { get; set; }
        public string? MetaDescription { get; set; }
        public string? MetaKeyWords { get; set; }
        public string? Slug { get; set; }

        public int? SgCategoryId { get; set; }
        public int? SgImageId { get; set; }
        public int? SgGalleryId { get; set; }
        public ICollection<ContentBlockForCreationDto>? ContentBlocks { get; set; }
    }
}

